content/shmem_request_allocate.tex

```diff
@@ -0,0 +1,34 @@
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void shmem_request_allocate(shmem_request_t *req, int flags);
+\end{Csynopsis}
+
+\begin{apiarguments}
+    \apiargument{IN/OUT}{req}{Handle to the request object.}
+    \apiargument{IN}{flags}{Options for the request creation.}
+\end{apiarguments}
+
+\apidescription{
+    This routine can be used to explicitly pre-allocate a request
+    object. On failure the handle will be set to SHMEM\_REQUEST\_INVALID.
+    The flags parameter can be used to pass hints to the
+    runtime.\\
```

Comment:
Do you want to define errors here, or just say it will return a negative integer on failure?